packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -86,9 +86,59 @@ class _RunMetrics {
   final int childCount;
 }
 
+/// How the child is inscribed into the available space.
+///
+/// See also:
+///
+///  * [RenderWrap], the wrap render object.
+///  * [Wrap], the wrap widget.
+///  * [Wrapped], the widget to set the [WrapFit].
+enum WrapFit {
+  /// The child is placed either in the current or the next run, depending on
+  /// its min intrinsic size in the [Wrap.direction].
+  /// Within this run, it is forced to fill the entire run, unless the [Wrap]
+  /// has no max size constraint in the run direction.
+  ///
+  /// This setting is more expensive, because it also computes the minimal
+  /// size of the child. Avoid using it for complex children.
+  ///
+  /// The [Wrapped] widget assigns this kind of [WrapFit] to its child.
+  runTight(true),
+
+  /// The child is placed either in the current or the next run, depending on
+  /// its min intrinsic size in the [Wrap.direction].
+  ///
+  /// This setting is more expensive, because it also computes the minimal
+  /// size of the child. Avoid using it for complex children.
+  runLoose(false),
+
+  /// The child is forced to fill the available space in a new run, unless the
+  /// [Wrap] has no max size constraint in the run direction.
+  tight(true),
+
+  /// The child can at most fill the available space in an empty run and is
+  /// placed base on its size and the remaining space.
+  ///
+  /// This is the default behavior for a child of [Wrap].
+  loose(false);
+
```

Comment:
I'm not sure about that. I think there are situations where it would be used. For example one might implement a custom widget that has an `WrapFit` option, that is then used in it's build method.

But I'd love to argue about that.

Reply:
For any situation where either the default `Wrap` behavior or one of these custom options should be allowed, we could just use a nullable `WrapFit?` parameter.

This would be analogous to `ColoredBox`, which has a required `color` parameter. Widgets like `DecoratedBox`, `Container`, and `Ink` have `color` as a nullable optional parameter.

Reply:
👍 I'll address this later. (In a few hours max)

Reply:
I just thought about that a bit more. Maybe `loose` should be a specific value. Here are some pros:
- I think for every enum option in the framework, there is one value for each situation.
- Usually, `null` means "not specified" or "not available".
- An user api that takes a `WrapFit?` could not use null to indicate the usage of the default.
- `switch(fit)` is easier to read.
- Docs might be easier to read.

Cons that I can think of:
- Users have one more options when doing "try and error programming".
- More docs needed.
- dead "code"?

"Do we want to prevent something like this?" is probably the core question.
```dart
Wrap(
  children: [
    Wrapped(   // <--- Unnecessary wrapper of the child
      fit: WrapFit.loose,
      child: SizedBox(),
    )]);

Reply:
I've thought about this a bit more as well.

Consider 2 enums in the framework:

```dart
enum Clip { none, hardEdge, antiAlias, antiAliasWithSaveLayer }

enum FlexFit { tight, loose }
```

For a while, I was puzzled as to why `Clip` has an option for "don't do any clipping", whereas both `FlexFit.tight` and `FlexFit.loose` are much different from not using a `Flexible` widget at all.

<br>

One difference is that `Clip` technically isn't part of the Flutter framework, but I believe that the main difference is: `Clip` applies to a widget with 1 child, whereas `FlexFit` can be applied to multiple children within a single `Row` or `Column`.

If an API wants to specify a default `clipBehavior`, it's pretty straightforward. However, specifying a default `FlexFit` is much more convoluted, since multiple `Flex` children could potentially have multiple different `FlexFit` settings.

The end result is that an API will generally either be opinionated about the `FlexFit` or allow the developer to pass their own `Flex` as a `child` widget.

<br>

Much like `Flex`, the `Wrap` widget has multiple children, which means that configuring a `WrapFit` will almost always be done directly, rather than via another API.

With this in mind, I still feel that the best move would be to remove `WrapFit.loose` to make things more simple.